comixed-batch/src/main/java/org/comixedproject/batch/comicbooks/processors/DeleteEmptyDirectoriesProcessor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * ComiXed - A digital comic book library management application.
+ * Copyright (C) 2022, The ComiXed Project
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses>
+ */
+
+package org.comixedproject.batch.comicbooks.processors;
+
+import java.io.File;
+import lombok.extern.log4j.Log4j2;
+import org.comixedproject.adaptors.file.FileAdaptor;
+import org.springframework.batch.item.ItemProcessor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * <code>DeleteEmptyDirectoriesProcessor</code> deletes a directory that contains no content.
+ *
+ * @author Darryl L. Pierce
+ */
+@Component
+@Log4j2
+public class DeleteEmptyDirectoriesProcessor implements ItemProcessor<File, Void> {
+  @Autowired private FileAdaptor fileAdaptor;
+
+  @Override
+  public Void process(final File directory) throws Exception {
+    if (directory.listFiles().length == 0) {
```

Comment:
Wouldn't this " ArrayUtils.isEmpty(directory.listFiles()) " be safer?

Reply:
I'm not sure, TBH. Is there a risk of File.listFiles() returning a null value? I had thought an instance of File will return an empty array if there are no contents. 

But, that said, I like your suggestion better. It makes it clear and readable the goal of the call. I'll update tonight and push the change you suggest.

Reply:
In the doc, they say:
Returns null if this abstract pathname does not denote a directory, or if an I/O error occurs.
But I was thinking of making the change for better readability of the purpose of the code.